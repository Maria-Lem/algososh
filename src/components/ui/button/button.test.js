import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import { Button } from './button';

describe('Testing button functionality', () => {
  it('Button is rendered with text', () => {
    const button = renderer.create(<Button text='Button text' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Button is rendered without text', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Button is disabled correctly', () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Button is using loading ion correctly', () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Clicking on button returns callback function', () => {
    const onClickEvent = jest.fn();
    render(<Button text='Button text' onClick={onClickEvent} />);
    fireEvent.click(screen.getByText('Button text'));
    expect(onClickEvent).toHaveBeenCalled();
  });
});
