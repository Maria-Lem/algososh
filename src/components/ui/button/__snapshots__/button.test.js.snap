// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button functionality Button is disabled correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing button functionality Button is rendered with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Button text
  </p>
</button>
`;

exports[`Testing button functionality Button is rendered without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing button functionality Button is using loading ion correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
